### PR TITLE
update build_once for cycle_gan

### PR DESCRIPTION
**Pull request by songyouwei**
- update **cycle_gan** to support remove build once
- use `Layer.sublayer.parameters()` instead of hard-coded name filters

### CODE REVIEW

File: dygraph/cycle_gan/model.py [hunk 1]
```diff
@@ -116,11 +123,13 @@ def forward(self,inputs):
...
+
 class build_gen_discriminator(fluid.dygraph.Layer):
-    def __init__(self,name_scope):
-        super(build_gen_discriminator,self).__init__(name_scope)
+    def __init__(self):
+        super(build_gen_discriminator, self).__init__()
         
-        self.conv0 = conv2d(self.full_name(),
+        self.conv0 = conv2d(
+            num_channels=3,
```
Comment (phlrain) [inline dygraph/cycle_gan/model.py]: why hard code

File: dygraph/cycle_gan/train.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def train(args):
         A_test_reader = data_reader.a_test_reader()
         B_test_reader = data_reader.b_test_reader()
 
-        cycle_gan = Cycle_Gan("cycle_gan", istrain=True)
+        cycle_gan = Cycle_Gan(istrain=True)
```
Comment (tink2123) [inline dygraph/cycle_gan/train.py]: add "input channel" ?

Comment (songyouwei) [reply dygraph/cycle_gan/train.py]: done

File: dygraph/cycle_gan/model.py (hunk 1 above)
Comment (songyouwei) [reply dygraph/cycle_gan/model.py]: changed to and `input_channel` arg

**Review by tink2123 — APPROVED** [review]
LGTM

**Review by phlrain — APPROVED** [review]

**Pull request merged**